src/source.js

```diff
@@ -34,6 +34,7 @@ async function openS3Stream (bucketRegion, url, car, retries = s3MaxRetries, ret
     try {
```

Comment:
I believe the `error = null` statement above needs to be placed inside the `try {}` block, and after the call to s3.  Doing so, should ensure that `error` is reset to `null` when s3 calls are successful on retries.  That in turn is necessary so that the `if (error)` condition is evaluated correctly.

Given S3 fails first attempt and error is set to a truthy value.
When S3 succeeds on subsequent attempt
Then error must be set to `null`